File: leetcode/solutions/0525.ContiguousArray/findMaxLength_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import findMaxLength from "./findMaxLength.ts";

Deno.test("0525. Contiguous Array", () => {
  type test = [number[], number];
  const tests: test[] = [
    [[0, 1], 2],
    [[0, 1, 0], 2],
    [[1, 0, 1], 2],
    [[1, 0, 1, 0], 4],
    [[1, 0, 1, 0, 0], 4],
    [[1, 1, 0, 1, 0], 4],
    [[1, 0, 1, 0, 1], 4],
    [[0, 0, 0, 1, 1, 1, 0], 6],
    [[0, 0, 1, 0, 0, 0, 1, 1], 6],
  ];

  for (const [arg, want] of tests) {
    assertEquals(findMaxLength(arg), want);
  }
});

File: leetcode/solutions/0525.ContiguousArray/findMaxLength_test.go
package solution

import (
	"testing"
)

func Test_0525_ContiguousArray(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  []int
		out int
	}{
		{[]int{0, 1}, 2},
		{[]int{0, 1, 0}, 2},
		{[]int{1, 0, 1}, 2},
		{[]int{1, 0, 1, 0}, 4},
		{[]int{1, 0, 1, 0, 0}, 4},
		{[]int{1, 1, 0, 1, 0}, 4},
		{[]int{1, 0, 1, 0, 1}, 4},
		{[]int{0, 0, 0, 1, 1, 1, 0}, 6},
		{[]int{0, 0, 1, 0, 0, 0, 1, 1}, 6},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := findMaxLength(tt.in)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}
